dashboard/routings/export_data.py

```diff
@@ -44,7 +45,63 @@ def submit_test_results():
         add_or_update_test(result['name'], time, result['successful'])
         add_test_result(result['name'], result['exec_time'], time, config.version, suite, result['iter'])
 
-    reset_tests_grouped()
-    add_tests_grouped(request.get_json()['grouped_tests'])
+    groups = request.get_json()['grouped_tests']
+    if groups:
+        reset_tests_grouped()
+        add_tests_grouped(groups)
 
     return '', 204
+
+
+@blueprint.route('/get_json_data/<security_token>', defaults={'time_from': 0})
+@blueprint.route('/get_json_data/<security_token>/<time_from>')
+def get_json_data_from(security_token: str, time_from: int):
+    """
+    Only get the data if the security token with the request is equivalent to the security token in the configuration.
+    :param security_token: security token to be specified.
+    :param time_from: (optional) if specified, only the data-values after this date are returned.
+                      input must be an timestamp value (utc) (= integer)
+    :return: all entries from the database. (endpoint-table)
+    """
+    if security_token == config.security_token:
+        data = []
+        try:
+            for entry in get_data_from(datetime.datetime.utcfromtimestamp(int(time_from))):
+                # nice conversion to json-object
+                data.append({
+                    'endpoint': entry.endpoint,
+                    'execution_time': entry.execution_time,
+                    'time': str(entry.time),
+                    'version': entry.version,
+                    'group_by': entry.group_by,
+                    'ip': entry.ip
+                })
+            return jsonify(data)
+        except ValueError as e:
+            return 'ValueError: {0}'.format(e)
+    return jsonify()
+
+
+@blueprint.route('/get_json_monitor_rules/<security_token>')
+def get_json_monitor_rules(security_token: str):
+    """
+    Only get the data if the security oken with the request is equivalent to the security token in the configuration.
+    :param security_token: security token for accessing the data
+    :return: all entries from the database (rules-table)
+    """
+    if security_token == config.security_token:
```

Comment:
if this security_token condition happens one more time... consider using a function_decorator to remove redundancy ;)